abb_bringup/launch/abb_moveit.launch.py

```diff
@@ -32,94 +30,87 @@ def launch_setup(context, *args, **kwargs):
     moveit_config_package = LaunchConfiguration("moveit_config_package")
     moveit_config_file = LaunchConfiguration("moveit_config_file")
 
-    # Planning context
-    robot_description_content = Command(
-        [
-            PathJoinSubstitution([FindExecutable(name="xacro")]),
-            " ",
-            PathJoinSubstitution(
-                [FindPackageShare(support_package), "urdf", robot_xacro_file]
-            ),
-        ]
-    )
-    robot_description = {"robot_description": robot_description_content}
-
-    robot_description_semantic_content = Command(
-        [
-            PathJoinSubstitution([FindExecutable(name="xacro")]),
-            " ",
-            PathJoinSubstitution(
-                [FindPackageShare(moveit_config_package), "config", moveit_config_file]
-            ),
-        ]
-    )
-    robot_description_semantic = {
-        "robot_description_semantic": robot_description_semantic_content.perform(
-            context
+    # MoveIt configuration
+    moveit_config = (
+        MoveItConfigsBuilder(
+            "abb_bringup", package_name=f"{moveit_config_package.perform(context)}"
         )
-    }
-
-    kinematics_yaml = load_yaml(
-        "abb_irb1200_5_90_moveit_config", "config/kinematics.yaml"
-    )
-
-    joint_limits_yaml = {
-        "robot_description_planning": load_yaml(
-            moveit_config_package.perform(context), "config/joint_limits.yaml"
+        .robot_description(
+            file_path=os.path.join(
+                get_package_share_directory(f"{support_package.perform(context)}"),
+                "urdf",
+                f"{robot_xacro_file.perform(context)}",
+            )
         )
-    }
-
-    # Planning Functionality
-    ompl_planning_pipeline_config = {
-        "move_group": {
-            "planning_plugin": "ompl_interface/OMPLPlanner",
-            "request_adapters": """default_planner_request_adapters/AddTimeOptimalParameterization default_planner_request_adapters/ResolveConstraintFrames default_planner_request_adapters/FixWorkspaceBounds default_planner_request_adapters/FixStartStateBounds default_planner_request_adapters/FixStartStateCollision default_planner_request_adapters/FixStartStatePathConstraints""",
-            "start_state_max_bounds_error": 0.1,
-        }
-    }
-    ompl_planning_yaml = load_yaml(
-        "abb_irb1200_5_90_moveit_config", "config/ompl_planning.yaml"
+        .robot_description_semantic(
+            file_path=os.path.join(
+                get_package_share_directory(
+                    f"{moveit_config_package.perform(context)}"
+                ),
+                "config",
+                f"{moveit_config_file.perform(context)}",
+            )
+        )
+        .planning_pipelines(
```

Comment:
Sorry one last thing came to mind. I don't think we should restrict the `planning_pipelines` to just `ompl`. If we remove this entry, the builder will load default configs for `ompl`, `stomp`, `chomp`and `pilz` which would be nicer for users to play around with in the rviz palnning panel.

Reply:
Just a quick question, I removed the lines here:

![Screenshot from 2024-01-03 12-57-20](https://github.com/PickNikRobotics/abb_ros2/assets/60989139/6a7bdf64-f75d-44c5-bb3c-0b8b0826a153)

But I was getting this error without adding a pilz_cartesian_limits.yaml file to the moveit_config package:

`[ERROR] [launch]: Caught exception in launch (see debug for traceback): "File /opt/robotic_platform/install/abb_irb1200_5_90_moveit_config/share/abb_irb1200_5_90_moveit_config/config/pilz_cartesian_limits.yaml doesn't exist"
`

So I added this to the abb_irb1200_5_90_moveit_config package in the config directory:

```yaml
cartesian_limits:
  max_trans_vel: 1.0
  max_trans_acc: 2.25
  max_trans_dec: -5.0
  max_rot_vel: 1.57
```

But even with this, If you select the pilz planner it doesn't seem to successfully plan a path.  The other planners I see available are ompl, and CHOMP and they can both successfully plan a path.  

Not sure how you would like me to proceed.

Reply:
I think the problem is because the `pilz_indsutrial_motion_planner` is loaded based on the deafult config file [in moveit_config_utils](https://github.com/ros-planning/moveit2/tree/main/moveit_configs_utils/default_configs) but there isn't a fefault `pilz_cartesian_limits.yaml` file. Hence you had to manually include on in the config folder here. I think if you use `moveit_setup_assistant`, it does generate this file in the local config.
Perhaps you can open an issue ticket in upstream moveit to include a default `pilz_cartesian_limits.yaml` file in `moveit_config_utils`?

> But even with this, If you select the pilz planner it doesn't seem to successfully plan a path. The other planners I see available are ompl, and CHOMP and they can both successfully plan a path.

I think this is because we have not set any acceleration limits in the `joint_limits.yaml` file which are needed by the planner. You should be seeing a message printout indicating this
```bash
[move_group-1] [INFO] [1704356296.402945412] [moveit_move_group_capabilities_base.move_group_capability]: Using planning pipeline 'pilz_industrial_motion_planner'
[move_group-1] [ERROR] [1704356296.402995752] [moveit.ros_planning.planning_pipeline]: Exception caught: 'acceleration limit not set for group manipulator'
```

If you edit the `joint_limits.yaml` file by setting `has_acceleration_limts: true` for each joint a specify some `max_acceleration` like `5.0`, you should be apply to plan PTP motions with `pilz_industrial_motion_planner`.

I think it's best to leave the acceleration limits as `false` for now until someone can share accurate values for this specific robot.